src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -138,8 +146,9 @@ await Task.WhenAny(_flushTask, Task.Delay(TimeSpan.FromSeconds(20)))
 
                 // In some situations, the flush thread can exit before flushing all the threads
                 // Force a flush for the leftover traces
-                completedTask = await Task.WhenAny(Task.Run(() => FlushBuffers(flushAllBuffers: true)), delay)
-                    .ConfigureAwait(false);
+                whenAnyArray[0] = Task.Factory.StartNew(state => ((AgentWriter)state).FlushBuffers(flushAllBuffers: true), this).Unwrap();
```

Comment:
I would switch back to `Task.Run`. `FlushAndCloseAsync` is called only once in the app lifetime, and the closure is negligible versus all the allocations from starting a new task. This is a case where, I believe, the added complexity and verbosity outweighs the benefits

Reply:
I hate DisplayClasses 🤣, will revert for readability.

Reply:
done in 40744c8